homeassistant/components/zoneminder.py

```diff
@@ -118,6 +122,22 @@ def _zm_request(method, api_url, data=None):
                           'decode "%s"', req.text)
 
 
+def zm_trigger(monitor, state, cause, duration):
+    """Attempt to connect over TCP socket and trigger recording."""
+    import socket
+    sock = socket.socket()
+    try:
+        sock.connect((ZM['host'], ZM['trigger_port']))
```

Comment:
This is crossing the boundary of protocol specific code that is allowed to be in Home Assistant. Please extract the Zoneminder logic into it's own Python library.